src/include/brain/brain.h

```diff
@@ -99,18 +133,25 @@ class Brain {
         scheduler_.RegisterPeriodicEvent(period, callback, job);
   }
 
-  inline void Run() {
-    scheduler_.EventLoop();
-  }
+  /**
+   * Executes the main eventloop on the brain. Tasks will begin executing periodically.
+   * Does not return unless there is an exception, or the loop is terminated.
+   */
+  inline void Run() { scheduler_.EventLoop(); }
 
-  inline void Terminate() {
-    scheduler_.ExitLoop();
-  }
+  /**
+   * Terminate the main event loop.
+   */
+  inline void Terminate() { scheduler_.ExitLoop(); }
 
  private:
+  // Main Event loop
   NotifiableTask scheduler_;
+  // collection of all the jobs registered
   std::unordered_map<std::string, BrainJob *> jobs_;
+  // mapping of jobs to their event structs (used to trigger the job)
   std::unordered_map<std::string, struct event *> job_handles_;
```

Comment:
I actually didn't see where this thing is used after the registration. Does this have some future use?

Reply:
Potentially. This is where you can drop a job or manually trigger it on demand,